lib/telemetry.go

```diff
@@ -210,6 +210,9 @@ type TelemetryConfig struct {
 	//
 	// hcl: telemetry { prometheus_retention_time = "duration" }
 	PrometheusOpts prometheus.PrometheusOpts
+
+	// ExtraSinks are additional metrics.MetricSink implementations that are to always be added.
+	ExtraSinks []metrics.MetricSink
```

Comment:
After discussion with @nickethier , the latest commit was modified hold this `ExtraSinks`, where the HCP sink will be appended.

This prevents the lib/telemetry library from having to know about the  OTEL/hcp sink.